app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -47,5 +47,17 @@ def create_backup_resource(type, id, data)
         {:task_id => cloud_volume.backup_create_queue(User.current_userid, data)}
       end
     end
+
+    def restore_backup_resource(type, id, data)
+      api_resource(type, id, "Restoring backup", :supports => :backup_restore) do |cloud_volume|
+        {:task_id => cloud_volume.backup_restore_queue(User.current_userid, data)}
```

Comment:
```suggestion
        {:task_id => cloud_volume.backup_restore_queue(User.current_userid, data.symbolize_keys)}
```

I think we need to do this.